test/renewLock.spec.ts

```diff
@@ -0,0 +1,193 @@
+    }
+    if (!process.env.QUEUE_NAME) {
+      throw new Error("Define QUEUE_NAME in your environment before running integration tests.");
+    }
+    if (!process.env.SUBSCRIPTION_NAME) {
+      throw new Error(
+        "Define SUBSCRIPTION_NAME in your environment before running integration tests."
+      );
+    }
+
+    namespace = Namespace.createFromConnectionString(process.env.SERVICEBUS_CONNECTION_STRING);
+    queueClient = namespace.createQueueClient(process.env.QUEUE_NAME);
+    topicClient = namespace.createTopicClient(process.env.TOPIC_NAME);
+    subscriptionClient = namespace.createSubscriptionClient(
+      process.env.TOPIC_NAME,
+      process.env.SUBSCRIPTION_NAME
+    );
+
+    const peekedQueueMsg = await queueClient.peek();
+    if (peekedQueueMsg.length) {
+      throw new Error("Please use an empty queue for integration testing");
+    }
+
+    const peekedSubscriptionMsg = await subscriptionClient.peek();
+    if (peekedSubscriptionMsg.length) {
+      throw new Error("Please use an empty Subscription for integration testing");
+    }
+  });
+
+  afterEach(async () => {
+    return namespace.close();
+  });
+
+  // Tests for Lock Renewal, see -  https://github.com/Azure/azure-service-bus-node/issues/103
+  // Receive a msg using Batch Receiver, test renewLock()
+  it("Queues - Receives a message using Batch Receiver renewLock() resets lock duration each time.", async function(): Promise<
+    void
+  > {
+    await testLockRenewalHappyCase(queueClient, queueClient);
+  });
+
+  it("TopicsSubs - Receives a message using Batch Receiver renewLock() resets lock duration each time.", async function(): Promise<
+    void
+  > {
+    await testLockRenewalHappyCase(topicClient, subscriptionClient);
+  });
+
+  // Receive a msg using Batch Receiver, wait until its lock expires, completing it now results in error
+  it("Queues - Receive a msg using Batch Receiver, wait until its lock expires, completing it now results in error", async function(): Promise<
+    void
+  > {
+    await testErrorOnLockExpiry(queueClient, queueClient);
+  });
+
+  it("TopicsSubs - Receive a msg using Batch Receiver, wait until its lock expires, completing it now results in error", async function(): Promise<
+    void
+  > {
+    await testErrorOnLockExpiry(topicClient, subscriptionClient);
+  });
+});
+
+async function testLockRenewalHappyCase(
+  senderClient: QueueClient | TopicClient,
+  receiverClient: QueueClient | SubscriptionClient
+): Promise<void> {
+  const currentExpectedLockExpiryTimeUtc = new Date();
+
+  await senderClient.sendBatch(testMessages);
+
+  const msgs = await receiverClient.receiveBatch(1);
+
+  should.equal(Array.isArray(msgs), true);
+  should.equal(msgs.length, 1);
+  should.equal(msgs[0].body, testMessages[0].body);
+  should.equal(msgs[0].messageId, testMessages[0].messageId);
+
+  console.log(`MessageLockedUntil: ${msgs[0].lockedUntilUtc}`);
```

Comment:
Console logs are generally not used in tests. Is there any reason you want to keep them?